--- a/log/logread.c
+++ b/log/logread.c
@@ -105,7 +105,8 @@
 	uint32_t p;
 	time_t t;
 	uint32_t t_ms = 0;
-	char *c, *m;
+	char *m;
+	char c[20];
 	int ret = 0;

 	if (sender.fd < 0)
@@ -124,7 +125,7 @@
 			rename(log_file, old);
 			free(old);
 		}
-		sender.fd = open(log_file, O_CREAT | O_WRONLY | O_APPEND, 0600);
+		sender.fd = open(log_file, O_CREAT | O_WRONLY | O_APPEND, 0644);
 		if (sender.fd < 0) {
 			fprintf(stderr, "failed to open %s: %m\n", log_file);
 			exit(-1);
@@ -141,9 +142,8 @@
 			snprintf(buf_ts, sizeof(buf_ts), "[%lu.%03u] ",
 					(unsigned long)t, t_ms);
 	}
-	c = ctime(&t);
 	p = blobmsg_get_u32(tb[LOG_PRIO]);
-	c[strlen(c) - 1] = '\0';
+	strftime(c, 20, "%Y-%m-%d %H:%M:%S", localtime(&t));

 	if (log_type == LOG_NET) {
 		int err;
@@ -270,7 +270,7 @@
  			uloop_timeout_set(&retry, 1000);
  		} else if (log_file) {
  			log_type = LOG_FILE;
- 			sender.fd = open(log_file, O_CREAT | O_WRONLY| O_APPEND, 0600);
+ 			sender.fd = open(log_file, O_CREAT | O_WRONLY| O_APPEND, 0644);
 	 		if (sender.fd < 0) {
 		 		fprintf(stderr, "failed to open %s: %m\n", log_file);
 			 	exit(-1);
