Index: httpd-2.4.25/docs/conf/httpd.conf.in
===================================================================
--- httpd-2.4.25.orig/docs/conf/httpd.conf.in
+++ httpd-2.4.25/docs/conf/httpd.conf.in
@@ -74,8 +73,8 @@ Listen @@Port@@
 # It is usually good practice to create a dedicated user and group for
 # running httpd, as with most system services.
 #
-User daemon
-Group daemon
+User nobody
+Group nogroup
 
 </IfModule>
 
@@ -411,8 +410,8 @@ Include @rel_sysconfdir@/extra/proxy-htm
 #       starting without SSL on platforms with no /dev/random equivalent
 #       but a statically compiled-in mod_ssl.
 #
-<IfModule ssl_module>
-SSLRandomSeed startup builtin
-SSLRandomSeed connect builtin
-</IfModule>
+#<IfModule ssl_module>
+#SSLRandomSeed startup builtin
+#SSLRandomSeed connect builtin
+#</IfModule>
 
