--- a/nss/coreconf/rules.mk
+++ b/nss/coreconf/rules.mk
@@ -259,9 +259,9 @@
 @$(MAKE_OBJDIR)
 	rm -f $@
 ifeq (,$(filter-out _WIN%,$(NS_USE_GCC)_$(OS_TARGET)))
-	$(AR) $(subst /,\\,$(OBJS))
+	$(AR) cr $@ $(subst /,\\,$(OBJS))
 else
-	$(AR) $(OBJS)
+	$(AR) cr $@ $(OBJS)
 endif
 	$(RANLIB) $@


--- a/nss/coreconf/UNIX.mk
+++ b/nss/coreconf/UNIX.mk
@@ -6,7 +6,7 @@
 XP_DEFINE  += -DXP_UNIX
 LIB_SUFFIX  = a
 DLL_SUFFIX  = so
-AR          = ar cr $@
+AR          = ar
 LDOPTS     += -L$(SOURCE_LIB_DIR)

 ifdef BUILD_OPT
