Description: Hack build routines to compile shared library
Author: loki_val and solar
Origin: http://sources.gentoo.org/viewcvs.py/gentoo-x86/net-libs/libsrtp/files/libsrtp-1.4.4-shared.patch
Last-Update: 2010-03-19
--- a/Makefile.in
+++ b/Makefile.in
@@ -15,7 +15,7 @@
 
 .PHONY: all test build_table_apps
 
-all: test 
+all: libsrtp.so.1 test
 
 runtest: build_table_apps test
 	@echo "running libsrtp test applications..."
@@ -67,7 +67,7 @@ libdir = @libdir@
 # implicit rules for object files and test apps
 
 %.o: %.c
-	$(COMPILE) -c $< -o $@
+	$(COMPILE) -fPIC -c $< -o $@
 
 %$(EXE): %.c
 	$(COMPILE) $(LDFLAGS) $< -o $@ $(SRTPLIB) $(LIBS)
